File: SlidingWindow/NumOfSubArrayWithAvgGreaThanEqToK/numOfSubarray.py
from typing import List

class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        n = len(arr)
        if k>n: return 0
        winSum = 0
        for i in range(k):
            winSum += arr[i]
        i = 0
        j = k
        ans = 0
        while j < n:
            winAvg = winSum // k
            if winAvg >= threshold: ans += 1
            winSum = (winSum - arr[i] + arr[j])
            i += 1
            j += 1
        if winSum//k >= threshold: ans += 1
        return ans

File: SlidingWindow/NumOfSubArrayWithAvgGreaThanEqToK/numOfSubarray.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int numOfSubarrays(vector<int>& arr, int k, int threshold) {
        int n = arr.size();
        if(k>n)return 0;
        int winSum = 0;
        for(int i=0;i<k;i++){
            winSum += arr[i];
        }
        int i=0;
        int j=k;
        int ans=0;
        while(j<n){
            int winAvg = winSum/k;
            if(winAvg>=threshold)ans++;
            winSum = winSum - arr[i] + arr[j];
            i++;
            j++;
        }
        if(winSum/k >= threshold)ans++;
        return ans;
    }
};